### PR TITLE
Reworks /proc/get_all_antagonists()

### PR DESCRIPTION
**Pull request by Mister-Moriarty**
[Internal] [Code Quality]

## About The PR:
Reworks `/proc/get_all_antagonists()` to iterate through a new `antagonists` global list, as opposed to iterating through every mind datum.
The `antagonists` global list will contain all antagonist datums sorted by role ID; i.e. all traitor datums will be filed under `antagonists["[ROLE_TRAITOR]"]`.

`/datum/game_mode/var/traitors` and `/datum/game_mode/var/Agimmicks` will eventually be superceded by `get_all_antagonists()`.
Deprecates `/proc/get_all_enemies()`.

## Why Is This needed?
Optimisation of the new system and removal of remnants from the old antagonist system that has been superceded by antagonist datums.

### CODE REVIEW

**Comment by Mister-Moriarty** [comment]
I do suppose `antagonists` could contain a list of antagonist datums, then antagonist minds can be accessed through the `owner` variable.

File: code/modules/antagonists/__antagonist.dm [hunk 1]
```diff
@@ -51,6 +51,10 @@ ABSTRACT_TYPE(/datum/antagonist)
 		src.pseudo = do_pseudo
 		src.vr = do_vr
 		if (!do_pseudo && !do_vr) // there is a special place in code hell for mind.special_role
+			if (!islist(antagonists["[src.id]"]))
+				antagonists["[src.id]"] = list()
+			antagonists["[src.id]"] += src
```
Comment (pali6) [inline code/modules/antagonists/__antagonist.dm:56]:
```suggestion
			LAZYLISTADD(antagonists["[src.id]"], src)
```
Seldom used but useful macro for this.

File: code/modules/antagonists/antagonist_helpers.dm [hunk 2]
```diff
@@ -1,25 +1,33 @@
...
-				break
+
+	if (antagonist_role_id)
+		return antagonists["[antagonist_role_id]"]
+
+	for (var/antagonist_type in concrete_typesof(/datum/antagonist))
+		var/datum/antagonist/A = antagonist_type
+		var/list/datum/antagonist/antagonist_roles = antagonists["[initial(A.id)]"]
+		if (length(antagonist_roles))
+			. += antagonist_roles
```
Comment (pali6) [inline code/modules/antagonists/antagonist_helpers.dm:18]: Am I missing something or could we just iterate over `antagonists` and join the values into one list?

Comment (Mister-Moriarty) [reply code/modules/antagonists/antagonist_helpers.dm]: Yes, that would be a far better method than the above.
Thank you!

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**